chia/cmds/wallet_funcs.py

```diff
@@ -132,415 +131,454 @@ async def get_unit_name_for_wallet_id(
+async def add_token(wallet_rpc_port: Optional[int], fp: Optional[int], asset_id: str, token_name: str) -> None:
+    async with get_wallet_client(wallet_rpc_port, fp) as (wallet_client, fingerprint, _):
+        if wallet_client is None:
+            return
+        try:
+            asset_id_bytes: bytes32 = bytes32.from_hexstr(asset_id)
+            existing_info: Optional[Tuple[Optional[uint32], str]] = await wallet_client.cat_asset_id_to_name(
+                asset_id_bytes
+            )
+            if existing_info is None or existing_info[0] is None:
+                response = await wallet_client.create_wallet_for_existing_cat(asset_id_bytes)
+                wallet_id = response["wallet_id"]
+                await wallet_client.set_cat_name(wallet_id, token_name)
+                print(f"Successfully added {token_name} with wallet id {wallet_id} on key {fingerprint}")
+            else:
+                wallet_id, old_name = existing_info
+                await wallet_client.set_cat_name(wallet_id, token_name)
+                print(
+                    f"Successfully renamed {old_name} with wallet_id {wallet_id} on key {fingerprint} to {token_name}"
+                )
+        except ValueError as e:
+            if "fromhex()" in str(e):
+                print(f"{asset_id} is not a valid Asset ID")
+            else:
+                raise
+
+
+async def make_offer(
+    *,
+    wallet_rpc_port: Optional[int],
+    fp: Optional[int],
+    d_fee: Decimal,
+    offers: Tuple[str],
+    requests: Tuple[str],
+    filepath: str,
+    reuse_puzhash: Optional[bool],
+) -> None:
+    async with get_wallet_client(wallet_rpc_port, fp) as (wallet_client, fingerprint, config):
+        if wallet_client is None:
+            return
+        fee: int = int(d_fee * units["chia"])
 
-    if [] in [offers, requests]:
-        print("Not creating offer: Must be offering and requesting at least one asset")
-    else:
-        offer_dict: Dict[Union[uint32, str], int] = {}
-        driver_dict: Dict[str, Any] = {}
-        printable_dict: Dict[str, Tuple[str, int, int]] = {}  # Dict[asset_name, Tuple[amount, unit, multiplier]]
-        royalty_asset_dict: Dict[Any, Tuple[Any, uint16]] = {}
-        fungible_asset_dict: Dict[Any, uint64] = {}
-        for item in [*offers, *requests]:
-            name, amount = tuple(item.split(":")[0:2])
-            try:
-                b32_id = bytes32.from_hexstr(name)
-                id: Union[uint32, str] = b32_id.hex()
-                result = await wallet_client.cat_asset_id_to_name(b32_id)
-                if result is not None:
-                    name = result[1]
-                else:
-                    name = "Unknown CAT"
-                unit = units["cat"]
-                if item in offers:
-                    fungible_asset_dict[name] = uint64(abs(int(Decimal(amount) * unit)))
-            except ValueError:
+        if len(offers) == 0 or len(requests) == 0:
```

Comment:
this is also an unrelated change (and I think a nicer form is `if offers == [] or requests == []`)

Reply:
mypy was mad and made be change it, but ill change it to that form.